src/dl_typelib_write_c_header.cpp

```diff
@@ -481,7 +532,7 @@ static void dl_context_write_c_header_enums( dl_binary_writer* writer, dl_ctx_t
 	free( enum_infos );
 }
 
-static void dl_context_write_type( dl_ctx_t ctx, dl_type_storage_t storage, dl_typeid_t tid, dl_binary_writer* writer )
+static dl_error_t dl_context_write_type( dl_ctx_t ctx, dl_type_storage_t storage, dl_typeid_t tid, dl_binary_writer* writer )
```

Comment:
This PR also dependended on [the ](https://github.com/wc-duck/datalibrary/pull/159), and it was to much work to rewrite all code to "not" do proper error handling. So now the error handling is so-so in the PR, and many lines like these are a duplication of that PR. Please try to ignore everything which deals with error propagation and indentation because of the changed return type in the function signature